trade-finance-manager-api/src/v1/__mocks__/api.js

```diff
@@ -180,14 +180,27 @@ module.exports = {
   getFacilityExposurePeriod: jest.fn(() => ({
     exposurePeriodInMonths: 12,
   })),
-  getPartyDbInfo: ({ companyRegNo }) =>
-    companyRegNo === 'NO_MATCH'
+  getPartyDbInfo: ({ companyRegNo }) => {
+    if (process.env.AUTOMATIC_SF_CUSTOMER_CREATION_ENABLED === 'true') {
+      return companyRegNo === 'NO_MATCH'
+        ? { status: 404, data: 'Party not found' }
+        : {
+            status: 200,
+            data: [
+              {
+                partyUrn: 'testPartyUrn',
+              },
+            ],
+          };
+    }
+    return companyRegNo === 'NO_MATCH'
       ? false
       : [
           {
             partyUrn: 'testPartyUrn',
           },
-        ],
+        ];
+  },
```

Comment:
The returns here are inconsistent, what's the reason for this please?

From line 196 we don't have `status` or `data` properties?

Reply:
So previously we didn't need to return a status code because the old GET method that checks for a customer existing in Salesforce via Informatica didn't distinguish between e.g. server error and a 404. In either case the BSG would have to check if a new customer had to be created, and create it if so. 

In the new implementation, we need to be able to tell if we receive a 404 when looking for the customer in Salesforce, because only in that case do we want to create a new customer. (We don't want to create a new one if we've just got a server error as the customer may well exist in Salesforce already).

Reply:
Thanks @natdeanlewissoftwire - seems to make sense.

What do you think about changing the function to be more if/else based? I know it's more lines but personally I think would be easier to read/maintain, e.g

```js
getPartyDbInfo: ({ companyRegNo }) => {
  const noCompanyMatch = companyRegNo === 'NO_MATCH';

  if (isAutomaticSalesforceCustomerCreationFeatureFlagEnabled()) {
    if (noCompanyMatch) {
      return { status: 404, data: 'Party not found' };
    }
    
    return {
      status: 200,
      data: [
        {
          partyUrn: 'testPartyUrn',
        },
      ],
    };
  } else {
    if (noCompanyMatch) {
      return false;
    }

    return {
      partyUrn: 'testPartyUrn',
    };
  }
}
```